src/prototype_creep_routing.js

```diff
@@ -162,32 +162,51 @@ Creep.prototype.getDirections = function(path, pathPos) {
   };
 };
 
+Creep.prototype.checkPreMove = function(datas) {
+  if (!datas.path) {
+    return true;
+  }
+  if (datas.pathPos < 0) {
+    this.memory.routing.pathPos = datas.pathPos;
+    return true;
+  }
+  return false;
+};
+
 Creep.prototype.followPath = function(action) {
   const route = this.getRoute();
   const routePos = this.getRoutePos(route);
 
   // TODO Disable base room for now
   // if (routePos === 0) {
-  // this.say('R:Base');
-  // return false;
+  //   this.say('R:Base');
+  //   return false;
   // }
 
   if (!this.memory.routing.targetId && this.room.name === this.memory.routing.targetRoom) {
     this.memory.routing.reached = true;
     return action(this);
   }
   const prepareData = this.moveByPathPrepare(route, routePos, 'pathStart', this.memory.routing.targetId);
-  if (prepareData.unit.preMove) {
-    if (prepareData.unit.preMove(this, prepareData.directions)) {
+  if (this.checkPreMove(prepareData)) {
+    if (prepareData.unit.preMove && prepareData.unit.preMove(this, prepareData.directions)) {
       return true;
     }
+    if (!prepareData.path || prepareData.path.length === 0) {
+      return false;
+    }
+  } else if (prepareData.unit.preMove && prepareData.unit.preMove(this, prepareData.directions)) {
```

Comment:
hmmm... I do not like this code duplication. Looking at it now, do we really need `checkPreMove`? We should run `preMove` both in case of `true` and `false`. Maybe it would be better to inline it? Or even move `this.memory.routing.pathPos = datas.pathPos` line into `moveByPathPrepare` function?

Reply:
I'm sorry I'm too annoying, even when it's my regression is being fixed)